### ISSUE TITLE
[aws-sns-sqs] fifo queue creation does not create the DLQ in the same type

### ISSUE THREAD

**Issue by flyingImer** [awslabs/aws-solutions-constructs#342]
DLQ was not created in the same type as the queue props specified, which, in turn, failed the CFN deployment later

### Reproduction Steps

```
new SnsToSqs(this, 'XXXXXX', {
      enableEncryptionWithCustomerManagedKey: false,
      existingTopicObj: someTopic,
      queueProps: {
        encryption: QueueEncryption.UNENCRYPTED,
        fifo: true,
      },
    });
```

### Error Log

When deploying with CFN, it failed with error message below:
```
Value {"deadLetterTargetArn":"xxxxxxxxx","maxReceiveCount":"15"} for parameter RedrivePolicy is invalid. Reason: Dead-letter queue must
be same type of queue as the source.
```

### Environment

  - **CDK CLI Version      :** 1.117.0
  - **CDK Framework Version:** 1.117.0
  - **AWS Solutions Constructs Version     :** 1.117.0
  - **OS                   :** macOS
  - **Language             :** ts

### Other

--- 

This is :bug: Bug Report

**Comment by biffgaut** [comment]
I think I see the problem, we'll take a crack at solving it. Thanks.

**Comment by biffgaut** [comment]
We started to code a fix to this, but as we dug in we realized it put us in a position where a customer was implicitly specifying incompatible values in two inputs (queueProps and deadLetterQueueProps), so the code was choosing between the customer's incompatible values for them. We weren't comfortable with that, nor with what meant going forward when there are so many situations where a customer could specify incompatible values across all constructs. Instead, we would prescribe explicitly specifying compatible values in the call:

```
new SnsToSqs(this, 'XXXXXX', {
  enableEncryptionWithCustomerManagedKey: false,
  existingTopicObj: newTopic,
  queueProps: {
    encryption: sqs.QueueEncryption.UNENCRYPTED,
    fifo: true,
  },
  deadLetterQueueProps: {
    fifo: true,
  },
}); 
```

That said, we are going to add a check for this situation to our input validation and provide a more explicit error message and solution when it occurs. We'll link that PR to this issue.
